File: cpp/643.cpp
#include <iostream>
#include <vector>
#include <string>
#include <numeric>
using namespace std;

class Solution {
public:
    double findMaxAverage(vector<int>& nums, int k) {
        int cnt = accumulate(nums.begin(), nums.begin() + k, 0);
        int maxCnt = cnt;
        for(int i = k; i < nums.size(); ++i) {
            cnt = cnt - nums[i - k] + nums[i];
            if (cnt > maxCnt) {
                maxCnt = cnt;
            }
        }
        return double(maxCnt) / k;
    }
};

File: python/643.py
class Solution(object):
    def findMaxAverage(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: float
        """
        cnt = sum(nums[:k])
        maxCnt = cnt
        for i in range(k, len(nums)):
            cnt = cnt - nums[i - k] + nums[i]
            if cnt > maxCnt:
                maxCnt = cnt
        return float(maxCnt) / k